PyStemmusScope/soil_io.py

```diff
@@ -0,0 +1,229 @@
+from pathlib import Path
+import hdf5storage
+import numpy as np
+import xarray as xr
+from . import variable_conversion as vc
+
+
+def _open_multifile_datasets(paths, lat, lon, lat_key='lat', lon_key='lon'):
+    """Internal function to open multifile netCDF files, and selects the lat & lon
+    before merging them by coordinates. xarray's open_mfdataset does not support this
+    type of functionality.
+
+    Args:
+        paths (iterable): Iterable containing the paths to the netCDF files
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        xarray.Dataset: Dataset containing the merged data for a single location in
+            space.
+    """
+    datasets = []
+    for file in paths:
+        ds = xr.open_dataset(file)
+        ds.attrs = '' # Drop attributed to avoid combine conflicts
+        datasets.append(ds.sel({lat_key: lat, lon_key: lon}, method='nearest'))
+    ds = xr.combine_by_coords(datasets)
+
+    return ds
+
+
+def _read_lambda_coef(lambda_directory, lat, lon):
+    """Internal function that reads the lambda coefficient files and returns the data
+    of interest in a dictionary.
+
+    Args:
+        lambda_directory (Path): Path to the directory which contains the lambda data.
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        dict: Dictionary containing the lambda coefficient data.
+    """
+    lambda_files = sorted(lambda_directory.glob("lambda_l*.nc"))
+
+    ds = _open_multifile_datasets(lambda_files, lat, lon)
+
+    # which depth indices the STEMMUS_SCOPE model expects
+    indices = [0, 2, 4, 5, 6, 7]
+    coef_lambda = ds['lambda'].isel(depth=indices).values
+
+    lambda_matfile = {'Coef_Lamda': coef_lambda}
+
+    return lambda_matfile
+
+
+def _read_soil_composition(soil_data_path, lat, lon):
+    """Internal function that reads the soil composition files and returns them in a
+    dictionary.
+
+    Args:
+        soil_data_path (Path): Path to the directory which contains the soil data.
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        dict: Dictionary containing the soil composition data.
+    """
+    soil_comp_fnames = ['CLAY1.nc', 'CLAY2.nc', 'OC1.nc', 'OC2.nc', 'SAND1.nc',
+                        'SAND2.nc', 'SILT1.nc', 'SILT2.nc']
+
+    soil_comp_paths = [soil_data_path / fname for fname in soil_comp_fnames]
+
+    ds = _open_multifile_datasets(soil_comp_paths, lat, lon)
+
+    depths_indices = [0, 2, 4, 5, 6, 7]
```

Comment:
Let's move ` depths_indices = [0, 2, 4, 5, 6, 7]` to `prepare_soil_data` and here consider it as an input argument to the function _read_soil_composition() as : _read_soil_composition(soil_data_path, lat, lon, depths_indices). We can add a ToDo to this module that the `depths_indices` can be read from config file.